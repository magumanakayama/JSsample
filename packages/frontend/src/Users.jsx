import { useUsers } from './useUsers';
import './App.css';

const User = ({ name }) => {
  return <li style={{padding: '8px'}}>{name}</li>;
};

const Users = () => {
  const { submit, input, inputText, users } = useUsers();

  const userList = users.map((user, index) => {
    return <User key={index} name={user}></User>;
  });

  const handleSubmit = (event) => {
    event.preventDefault();
    submit();
  };
  
  const handleChange = (event) => {
    input(event.target.value);
  };

  return (
    <div className="App">
      <ul>
        {userList}
      </ul>
      <form onSubmit={(e) => handleSubmit(e)}>
        <input type = "text" onChange={handleChange}/>
        <button type="submit">追加</button>
      </form>
      <div>入力値：{inputText}</div>
    </div>
  );
}

export default Users;
