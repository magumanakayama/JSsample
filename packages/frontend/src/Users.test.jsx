import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Users from './Users';
import { it } from 'vitest';


// const server = setupServer();

// beforeAll(() => server.listen());
// afterEach(() => server.resetHandlers());
// afterAll(() => server.close());


// submitとinputをグローバルで定義
const submit = vi.fn();
const input = vi.fn();

vi.mock('./useUsers', () => ({
  useUsers: () => ({
    submit,
    input,
    inputText: '',
    users: ['maguma', 'nakayama', 'makise'],
  }),
}));

const listExpecter = async (assertList) => {
  for (const item of assertList) {
    await waitFor(() => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });
  }
};

describe('Usersコンポーネントのテスト', () => {
  it('初期描画のテスト', async () => {
    render(<Users />);
    await listExpecter(['maguma', 'nakayama', 'makise']);
  });

  it('追加ボタン押下でsubmitが呼ばれる', async () => {
    render(<Users />);
    fireEvent.submit(screen.getByRole('button', { name: '追加' }));
    expect(submit).toHaveBeenCalledTimes(1);
  });

  it('入力値が更新される', async () => {
    render(<Users />);
    const inputElement = screen.getByRole('textbox');
    fireEvent.change(inputElement, { target: { value: '新しいユーザー' } });
    expect(input).toHaveBeenCalledWith('新しいユーザー');
  });

  it('Usersのスナップショットテスト', async () => {
    const { container } = render(<Users />);
    expect(container).toMatchSnapshot();
  });
});