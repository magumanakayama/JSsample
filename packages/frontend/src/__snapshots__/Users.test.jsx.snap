// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Usersコンポーネントのテスト > Usersのスナップショットテスト 1`] = `
<div>
  <div
    class="App"
  >
    <ul>
      <li
        style="padding: 8px;"
      >
        maguma
      </li>
      <li
        style="padding: 8px;"
      >
        nakayama
      </li>
      <li
        style="padding: 8px;"
      >
        makise
      </li>
    </ul>
    <form>
      <input
        type="text"
      />
      <button
        type="submit"
      >
        追加
      </button>
    </form>
    <div>
      入力値：
    </div>
  </div>
</div>
`;
